File: src/java/org/javaturk/dp/ch04/abstractFactory/gui/sol2/GUIFactoryImpl.java
package org.javaturk.dp.ch04.abstractFactory.gui.sol2;

public class GUIFactoryImpl implements GUIFactory {
    public Button createButton() {
        return new Button();
    }

	@Override
	public List createList() {
		return new List();
	}

	@Override
	public Table createTable() {
		return new Table();
	}
}

File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/abstract-factory/gui/sol2/gui-factory-impl.ts
import {GUIFactory} from "./gui-factory";
import {Button} from "../sol1/button";
import {List} from "../sol1/list";
import {Component} from "../sol1/component";
import {Table} from "../sol1/table";

export class GUIFactoryImpl implements GUIFactory {

  createButton(): Component {
    return new Button();
  }

  createList(): Component {
    return new List();
  }

  createTable(): Component {
    return new Table();
  }
}